app/assets/javascripts/initializers/initializeAllFollowButts.js

```diff
@@ -1,10 +1,54 @@
 function initializeAllFollowButts() {
   var followButts = document.getElementsByClassName('follow-action-button');
   for (var i = 0; i < followButts.length; i++) {
-    initializeFollowButt(followButts[i]);
+    if (!followButts[i].className.includes("follow-user")) {
+      initializeFollowButt(followButts[i]);
+    };
   }
 }
 
+function fetchUserFollowStatuses(idButtonHash) {
+  var idString = "";
+  Object.keys(idButtonHash).forEach(function(id) {
+    idString += ("ids[]=" + id + "&");
+  });
+
+  fetch(`/follows/bulk_show?${idString}followable_type=User`, {
+    method: 'GET',
+    headers: {
+      Accept: 'application/json',
+      'X-CSRF-Token': window.csrfToken,
+      'Content-Type': 'application/json',
+    },
+    credentials: 'same-origin',
+  }).then((response) => response.json())
+    .then((idStatuses) => {
+      Object.keys(idStatuses).forEach(function(id) {
+        addButtClickHandle(idStatuses[id], idButtonHash[id]);
+      })
+    });
+}
+
+function initializeUserFollowButtons(buttons) {
+  var userIds = {};
```

Comment:
maybe we can just exit this function if `buttons.length` is 0, to avoid the check at line 44

Reply:
We still need the 44 check bc we might add the `addModalEventListener(buttons[i]);` to buttons if the user is logged out.